nexus/db-queries/src/db/datastore/v2p_mapping.rs

```diff
@@ -7,34 +7,132 @@ use crate::context::OpContext;
 use omicron_common::api::external::ListResultVec;
 
 impl DataStore {
     pub async fn v2p_mappings(
         &self,
         opctx: &OpContext,
     ) -> ListResultVec<V2PMappingView> {
-        use db::schema::v2p_mapping_view::dsl;
+        use db::schema::instance::dsl as instance_dsl;
+        use db::schema::network_interface::dsl as network_interface_dsl;
+        use db::schema::probe::dsl as probe_dsl;
+        use db::schema::sled::dsl as sled_dsl;
+        use db::schema::vmm::dsl as vmm_dsl;
+        use db::schema::vpc::dsl as vpc_dsl;
+        use db::schema::vpc_subnet::dsl as vpc_subnet_dsl;
+
+        use db::schema::network_interface;
 
         opctx.check_complex_operations_allowed()?;
 
         let mut mappings = Vec::new();
         let mut paginator = Paginator::new(SQL_BATCH_SIZE);
         while let Some(p) = paginator.next() {
-            let batch = paginated(
-                dsl::v2p_mapping_view,
-                dsl::nic_id,
-                &p.current_pagparams(),
-            )
-            .select(V2PMappingView::as_select())
-            .load_async(&*self.pool_connection_authorized(opctx).await?)
-            .await
-            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+            let batch: Vec<_> =
+                paginated(
+                    network_interface_dsl::network_interface,
+                    network_interface_dsl::id,
+                    &p.current_pagparams(),
+                )
+                .inner_join(
+                    instance_dsl::instance
+                        .on(network_interface_dsl::parent_id
+                            .eq(instance_dsl::id)),
+                )
+                .inner_join(vmm_dsl::vmm.on(
+                    vmm_dsl::id.nullable().eq(instance_dsl::active_propolis_id),
+                ))
+                .inner_join(vpc_subnet_dsl::vpc_subnet.on(
+                    vpc_subnet_dsl::id.eq(network_interface_dsl::subnet_id),
+                ))
+                .inner_join(
+                    vpc_dsl::vpc
+                        .on(vpc_dsl::id.eq(network_interface_dsl::vpc_id)),
+                )
+                .inner_join(
+                    sled_dsl::sled.on(sled_dsl::id.eq(vmm_dsl::sled_id)),
+                )
+                .filter(network_interface::time_deleted.is_null())
+                .filter(
+                    network_interface::kind.eq(NetworkInterfaceKind::Instance),
+                )
+                .sled_filter(SledFilter::V2PMapping)
+                .select((
+                    NetworkInterface::as_select(),
+                    Sled::as_select(),
+                    Vpc::as_select(),
+                ))
+                .load_async(&*self.pool_connection_authorized(opctx).await?)
+                .await
+                .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?
+                .into_iter()
+                .map(|i: (NetworkInterface, Sled, Vpc)| V2PMappingView {
```

Comment:
Tiny nit - I think since these have fields with the same name (`.identity`, `.ip`) I'd probably give these full names instead of `i.N`:

```suggestion
                .map(|(nic, sled, vpc): (NetworkInterface, Sled, Vpc)| V2PMappingView {
```

(same comment in the next `paginated` block below)